docs/how-to-guides/develop-a-tool/create-cascading-tool-inputs.md

```diff
@@ -0,0 +1,136 @@
+# Creating Cascading Tool Inputs
+
+Cascading input settings are useful when the value of one input field determines which subsequent inputs are shown. This makes the input process more streamlined, user-friendly, and error-free. This guide will walk through how to create cascading inputs for your tools.
+
+## Prerequisites
+Please make sure you have the latest version of [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) installed (v1.2.0+).
+
+
+## Create a tool with cascading inputs
+We'll build out an example tool to show how cascading inputs work. The `student_id` and `teacher_id` inputs will be controlled by the value selected for the `user_type` input. Here's how to configure this in the tool code and YAML.
+
+1. Develop the tool function, following the [cascading inputs example](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_cascading_inputs.py). Key points:
+    * Use the `@tool` decorator to mark the function as a tool.
+    * Define `UserType` as an Enum class, as it accepts only a specific set of fixed values in this example. In fact, you can use other types depending on your own scenarios.
+    * Conditionally use inputs in the tool logic based on `user_type`.
+
+```python
+from enum import Enum
+
+from promptflow import tool
+
+
+class UserType(str, Enum):
+    STUDENT = "student"
+    TEACHER = "teacher"
+
+
+@tool
+def my_tool(user_type: Enum, student_id: str = "", teacher_id: str = "") -> str:
+    """This is a dummy function to support cascading inputs.
+
+    :param user_type: user type, student or teacher.
+    :param student_id: student id.
+    :param teacher_id: teacher id.
+    :return: id of the user.
+    If user_type is student, return student_id.
+    If user_type is teacher, return teacher_id.
+    """
+    if user_type == UserType.STUDENT:
+        return student_id
+    elif user_type == UserType.TEACHER:
+        return teacher_id
+    else:
+        raise Exception("Invalid user.")
+```
+
+2. Generate a starting YAML for your tool per the [tool package guide](create-and-use-tool-package.md), then update it to enable cascading:
+
+    Add `enabled_by` and `enabled_by_value` to control visibility of dependent inputs. See the [example YAML](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/yamls/tool_with_cascading_inputs.yaml) for reference.
+
+    * The `enabled_by` attribute specifies which input field controls the visibility of the dependent input field.
```

Comment:
The enabled_by attribute specifies the input field, which must be an enum type, that controls the visibility of the dependent input field.

The enabled_by_value attribute defines the accepted enum values from the enabled_by field that will make this dependent input field visible.